src/main/java/com/example/onlinebookstore/service/impl/CategoryServiceImpl.java

```diff
@@ -47,13 +48,17 @@ public CategoryDto create(CreateCategoryRequestDto requestDto) {
 
     @Override
     public CategoryDto updateById(CreateCategoryRequestDto requestDto, Long id) {
+        categoryRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find category by id: " + id));
         Category model = categoryMapper.toModel(requestDto);
         model.setId(id);
         return categoryMapper.toDto(categoryRepository.save(model));
     }
 
     @Override
     public void deleteById(Long id) {
+        categoryRepository.findById(id)
```

Comment:
same here. Also usually when deleting a resource with non-existent ID is considered a valid case and exception is not thrown